aptos-move/framework/aptos-framework/sources/event.move

```diff
@@ -54,4 +54,23 @@ module aptos_framework::event {
     public fun destroy_handle<T: drop + store>(handle: EventHandle<T>) {
         EventHandle<T> { counter: _, guid: _ } = handle;
     }
+
+    #[test_only]
+    use std::option::{Self, Option};
+
+    #[test_only]
+    public fun emitted_event<T: drop + store>(handle: &EventHandle<T>, sequence_number: u64): Option<T> {
+        let id = bcs::to_bytes(&handle.guid);
+        if (emitted_internal<T>(id, sequence_number)) {
+            option::some(emitted_event_internal<T>(id, sequence_number))
+        } else {
+            option::none()
+        }
+    }
+
+    #[test_only]
+    native fun emitted_internal<T: drop + store>(id: vector<u8>, sequence_number: u64): bool;
```

Comment:
How about a public test-only wrapper around this `was_event_emitted()`?

Reply:
you mean so you don't have to write the check and assert yourself?

Reply:
Sure, the same functionality could be achieved within each unit context by checking the result of `event::emitted_event()` as some/none,  but I figure that would be a common function call that could be abstracted into a helper

Not super necessary, just requiring less function calls for test writers

Reply:
addressed